addons/sale/models/sale_order.py

```diff
@@ -32,6 +32,13 @@ def _get_default_require_signature(self):
     def _get_default_require_payment(self):
         return self.env.company.portal_confirmation_pay
 
+    def _get_amount_paid_with_gift_card(self):
```

Comment:
Actually we can move `_compute_amount_total_without_delivery` to `sale` instead of making a whole new function.